### ISSUE TITLE
prefer-const should recognise the object-path library making changes to a variable

### ISSUE THREAD

**Issue by revelt** [eslint/eslint#9600]
**What rule do you want to change?**
`prefer-const`

**Does this change cause the rule to produce more or fewer warnings?**
same or less

**How will the change be implemented? (New option, new default behavior, etc.)?**
parser should be improved to recognise variables being changed via [object-path](https://www.npmjs.com/package/object-path)

**Please provide some example code that this change will affect:**

`prefer-const` rule currently does not detect that `objectPath.set()` method changes the variable and sets the original declaration to be `const`.

```js
let a = {}
objectPath.set(a, 'somekey', 'someval')
```

ESLint would currently auto-correct this to 

```js
const a = {}
objectPath.set(a, 'somekey', 'someval')
```

Around 100K people download `object-path` a-day, so it's quite popular utility library (https://npmcharts.com/compare/object-path) and therefore it's worth adapting ESLint to recognise its methods.

**Comment by platinumazure** [comment]
This would be an excellent candidate for creating a plugin. I'm :-1: to adding to core since this isn't a language syntax issue or ECMA-standard library feature.

My recommendation would be to use [eslint-rule-composer](https://github.com/not-an-aardvark/eslint-rule-composer) to create a custom `prefer-const` rule that also takes objectPath into account, and then distributing that rule in a plugin.

**Comment by revelt** [comment]
@platinumazure, it makes sense. Should we close this issue or should we keep it for a public record until `objectPath` is taken care of? Feel free to close if you feel so.

**Comment by platinumazure** [comment]
I'm happy to let other team members weigh in in case they disagree with my assessment, but you may close if you think my assessment is correct. Whether the issue is closed or not, you would be welcome to come back here and comment once the plugin is created. (I don't think we should keep it open specifically while waiting for that plugin, though. Closed issues are searchable too!)

**Comment by revelt** [comment]
logical. thank you!
